packages/artifact/__tests__/upload-specification.test.ts

```diff
@@ -0,0 +1,340 @@
+        )
+      } else if (specification.absoluteFilePath === goodItem3Path) {
+        expect(specification.uploadFilePath).toEqual(
+          path.join(artifactName, 'folder-d', 'good-item3.txt')
+        )
+      } else if (specification.absoluteFilePath === goodItem4Path) {
+        expect(specification.uploadFilePath).toEqual(
+          path.join(artifactName, 'folder-d', 'good-item4.txt')
+        )
+      } else if (specification.absoluteFilePath === goodItem5Path) {
+        expect(specification.uploadFilePath).toEqual(
+          path.join(artifactName, 'good-item5.txt')
+        )
+      } else if (specification.absoluteFilePath === extraFileInFolderCPath) {
+        expect(specification.uploadFilePath).toEqual(
+          path.join(
+            artifactName,
+            'folder-a',
+            'folder-b',
+            'folder-c',
+            'extra-file-in-folder-c.txt'
+          )
+        )
+      } else if (specification.absoluteFilePath === amazingFileInFolderHPath) {
+        expect(specification.uploadFilePath).toEqual(
+          path.join(artifactName, 'folder-h', 'amazing-item.txt')
+        )
+      } else {
+        throw new Error('this should never be reached')
+      }
+    }
+  })
+
+  it('Upload Specification - Directories should not be included', async () => {
+    const folderEPath = path.join(root, 'folder-a', 'folder-b', 'folder-e')
+    const filesWithDirectory = [
+      goodItem1Path,
+      goodItem4Path,
+      folderEPath,
+      badItem3Path
+    ]
+    const specifications = getUploadSpecification(
+      artifactName,
+      root,
+      filesWithDirectory
+    )
+    expect(specifications.length).toEqual(3)
+    const absolutePaths = specifications.map(item => item.absoluteFilePath)
+    expect(absolutePaths.includes(goodItem1Path)).toEqual(true)
+    expect(absolutePaths.includes(goodItem4Path)).toEqual(true)
+    expect(absolutePaths.includes(badItem3Path)).toEqual(true)
+
+    for (const specification of specifications) {
+      if (specification.absoluteFilePath === goodItem1Path) {
+        expect(specification.uploadFilePath).toEqual(
+          path.join(
+            artifactName,
+            'folder-a',
+            'folder-b',
+            'folder-c',
+            'good-item1.txt'
+          )
+        )
+      } else if (specification.absoluteFilePath === goodItem2Path) {
+        expect(specification.uploadFilePath).toEqual(
+          path.join(artifactName, 'folder-d', 'good-item2.txt')
+        )
+      } else if (specification.absoluteFilePath === goodItem4Path) {
+        expect(specification.uploadFilePath).toEqual(
+          path.join(artifactName, 'folder-d', 'good-item4.txt')
+        )
+      } else if (specification.absoluteFilePath === badItem3Path) {
+        expect(specification.uploadFilePath).toEqual(
+          path.join(artifactName, 'folder-f', 'bad-item3.txt')
+        )
+      } else {
+        throw new Error('this should never be reached')
```

Comment:
If we're going to fail the test by throwing an error, we should be more explicit about what went wrong.

In this specific case, it might be better to simplify this into an array contains
```js
expect(badUploadFilePaths).toContain(specification.uploadFilePath)
```
https://jestjs.io/docs/en/expect#tocontainitem

Reply:
`toContainer` is fairly similar to what I have with `includes`, I can swap that out in a few places that I use it because it seems nicer

I use an array to check for the `absolutePaths` however using just another array for `uploadFilePath` does not 100% check for what we need. It is possible that all the correct strings exist for `uploadFilePath` in an array, however the right combination must match with the `absolutePath` which is what I'm doing with the `if else-if` statements